cpp/dolfinx/fem/interpolate.h

```diff
@@ -510,9 +514,10 @@ void interpolate(Function<T>& u, xt::xarray<T>& f,
                           xt::xall<std::size_t>>;
     auto pull_back_fn = element->map_fn<U_t, U_t, J_t, J_t>();
 
-    for (std::int32_t c : cells)
+    for (std::size_t c = 0; c < cells.size(); ++c)
```

Comment:
Why have the loops been changed from range-based?

Reply:
Because you need to access both c and cells[c] (see [line 547](https://github.com/FEniCS/dolfinx/blob/d2add3693f50c8ba798e1ff8b8368757115de453/cpp/dolfinx/fem/interpolate.h#L547))
The code in main does not work if you send in a set of cells not equal to the local range.
Changing the loop here fixes this bug

Reply:
Similar comment is relevant for 453 and the first for at 446